### ISSUE TITLE
dmg images are unsigned

### ISSUE THREAD

**Issue by zebambam** [ZcashFoundation/zecwallet#197]
Mac software releases are currently unsigned, meaning that users need to understand how to undermine os x signing process - which they've made deliberately convoluted (and won't actually work on some corporate builds).

I just had a conversation with someone internally here about how we probably shouldn't put out instructions to users, some of whom have no idea when this might or might not be appropriate to do, how to launch unsigned apps. We'd love to help you to bring your software to a wider audience, but at the moment it's hard to do that without also teaching those folks to possibly shoot themselves in the foot later on.

The solution would be to sign up for an Apple developer account to receive a certificate and then add a signing step to the build process that signs the release with the certificate. The cost of an account is 99 USD which we can bankroll if its an issue and I can even give you a PR if you point me in the right spot for your build stuff.

**Comment by zebambam** [comment]
Update: a 15 second search found it here: https://github.com/ZcashFoundation/zecwallet/blob/master/src/scripts/mkmacdmg.sh :)

**Comment by adityapk00** [comment]
Is this the same account as an iOS appstore developer account? I already have that, so I can maybe reuse that account. I don't know how to do the actual signing, so if you send me a PR, that will be enormously helpful.

The code that builds the Mac DMG is here: 
https://github.com/ZcashFoundation/zecwallet/blob/master/src/scripts/mkmacdmg.sh#L91

**Comment by zebambam** [comment]
Yeah now that I'm looking at it I'm half-convinced that it's impossible, but I'll keep looking at it.

**Comment by zebambam** [comment]
This thread seems to think that a change to macdeployqt will do it.. https://forum.qt.io/topic/80609/how-to-sign-a-qt-app-with-a-certificate-in-osx/4

I'll try it now.

**Comment by zebambam** [comment]
@adityapk00 what's your apple developer account id? I think it's just a case of adding 
`-codesign=<id>`
to the macdeployqt line.

**Comment by zebambam** [comment]
Ah, so the id is the name of the cert, and you download that from https://developer.apple.com/account/ios/certificate/

**Comment by adityapk00** [comment]
It looks like I only have an iOS certificate. I'll try it now...

**Comment by adityapk00** [comment]
There's all these certificate types. Do you know which one I need?

![image](https://user-images.githubusercontent.com/31996805/67974932-81258580-fbd0-11e9-99f8-58e6c6eb05e9.png)

**Comment by adityapk00** [comment]
I'm getting all kinds of errors. 
```
ERROR: Codesign signing error:
ERROR: "/Volumes/macminie/prod/zecwallet/zecwallet.app/Contents/MacOS/zecwallet: code object is not signed at all\nIn subcomponent: /Volumes/macminie/prod/zecwallet/zecwallet.app/Contents/Frameworks/QtCore.framework\n"
ERROR: codesign verification error:
ERROR: "zecwallet.app: code object is not signed at all\nIn architecture: x86_64\n"
```

I'll keep trying

**Comment by zebambam** [comment]
ACK, googling..

**Comment by zebambam** [comment]
Allegedly it's necessary to sign each framework individually https://bugreports.qt.io/browse/QTBUG-32896?focusedCommentId=251937&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel

**Comment by zebambam** [comment]
But I'm not expecting you to work from these flimsy instructions. I've started installing qt5 so I can build it myself.

**Comment by zebambam** [comment]
Hey, sorry I have to step back on this now. It's not super urgent but it would of course be nice for the reasons outlined at the top. I may be able to take a look next week. Meantime I'll raise a ticket and put it on our internal backlog to revisit this.

**Comment by adityapk00** [comment]
Fixed with 0.8.0
